Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1041,6 +1042,14 @@ BoardInit (
   case PrePciEnumeration:
     break;
   case PostPciEnumeration:
+    if (GetPayloadId () == UEFI_PAYLOAD_ID_SIGNATURE && GetBootMode() == BOOT_ON_S3_RESUME) {

```

Comment:
You don't have to check UEFI payload.  If it is some other payload which has SMI initialized, the same should be applied.

Reply:
Yeah I now check for S3 only